swap/tests/refund_restart.rs

```diff
@@ -0,0 +1,162 @@
+use crate::testutils::{init_alice, init_bob};
+use futures::future::try_join;
+use libp2p::Multiaddr;
+use rand::rngs::OsRng;
+use swap::{alice, alice::swap::AliceState, bob, bob::swap::BobState, storage::Database};
+use tempfile::tempdir;
+use testcontainers::clients::Cli;
+use testutils::init_tracing;
+use uuid::Uuid;
+use xmr_btc::{bitcoin, config::Config};
+
+pub mod testutils;
+
+// Bob locks btc and Alice locks xmr. Alice fails to act so Bob refunds. Alice
+// then also refunds.
+#[tokio::test]
+async fn both_refund() {
```

Comment:
This only tests for Alice to restart and refund. We need a test for Bob to restart to ensure that database save/restore works as expected.